packages/devtools-connect/src/connect.ts

```diff
@@ -588,10 +588,15 @@ async function connectMongoClientImpl({
       ca ? { ca } : {},
     );
 
+    const customLookup = mongoClientOptions.lookup;
     // Adopt dns result order changes with Node v18 that affected the VSCode extension VSCODE-458.
     // Refs https://github.com/microsoft/vscode/issues/189805
     mongoClientOptions.lookup = (hostname, options, callback) => {
-      return dns.lookup(hostname, { verbatim: false, ...options }, callback);
+      return (customLookup ?? dns.lookup)(
+        hostname,
+        { verbatim: false, ...options },
+        callback,
+      );
```

Comment:
```suggestion
// Adopt dns result order changes with Node v18 that affected the VSCode extension VSCODE-458.
// Refs https://github.com/microsoft/vscode/issues/189805
mongoClientOptions.lookup ??= (hostname, options, callback) => {
      return dns.lookup(hostname, { verbatim: false, ...options }, callback);
```

This could also just be `??=` if we want to use whatever lookup we are given, but it seems more correct to me that this package owns its opinion on the verbatim flag